examples/mnist/train.cpp

```diff
@@ -37,25 +37,23 @@ static void construct_net(network<sequential> &nn,
   nn << convolutional_layer(32, 32, 5, 1,
                             6,  // C1, 1@32x32-in, 6@28x28-out
                             padding::valid, true, 1, 1, backend_type)
-     << tanh_layer(28, 28, 6)
-     << average_pooling_layer(28, 28, 6,
-                              2)  // S2, 6@28x28-in, 6@14x14-out
-     << tanh_layer(14, 14, 6)
+     << tanh_layer() << average_pooling_layer(28, 28, 6,
+                                              2)  // S2, 6@28x28-in, 6@14x14-out
+     << tanh_layer()
```

Comment:
it can be solved by reducing the variables name size, but yeah that's another war